server/game/cards/01-Core/MessengerRaven.js

```diff
@@ -1,21 +1,17 @@
 const DrawCard = require('../../drawcard.js');
 
 class MessengerRaven extends DrawCard {
-    setupCardAbilities() {
+    setupCardAbilities(ability) {
         this.action({
-            title: 'Return Messanger Raven to your hand and draw 1 card',
-            method: 'return',
-            phase: 'dominance'
-        });
-    }
-
-    return(player) {
-        player.moveCard(this, 'hand');
-        player.drawCardsToHand(1);
+            title: 'Draw 1 card',
+            phase: 'dominance',
+            cost: ability.costs.returnSelfToHand(),
+            handler: context => {
+                context.player.drawCardsToHand(1);
 
-        this.game.addMessage('{0} returns {1} to their hand and draws 1 card', player, this);
-
-        return true;
+                this.game.addMessage('{0} returns {1} to their hand and draws 1 card', context.player, this);
```

Comment:
minor nit, this should probably be "{0} returns {1} to their hand *to* draw 1 card"